fledge-iot/fledge
# Delete OTT tokens for user if something is updated for it. FOGL-6892.

We should delete the ott token if following cases happen:
Cases

1. The user is logout from admin's user management. 
2. The user logout itself.
3. The user is deleted. 
4. The password of user is changed through admin. 
5. The auth method for user is changed  through admin. 
6. The password of user is changed through self. 
7. The auth method for user is changed  through self. 

How to Test 

1. Login through Admin .
2. create a dummy user. 
3. In another tab (in cognito) login thorough this user. 
4. Obtain the auth token from the other tab. 
5. Run the command 
```
curl -H "authorization: <token>" -skX GET https://localhost:1995/fledge/auth/ott | jq

```

6. Perform any of the operations above . 
7. Run the command 

```
curl -d '{"ott": "ott_token"}' -skX POST https://localhost:1995/fledge/login

```

8 .expected reponse 
`401: Authentication failed. Either the given token expired or already used.`
